hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/io/storage/row/HoodieRowParquetWriteSupport.java

```diff
@@ -73,6 +189,180 @@ public void add(UTF8String recordKey) {
         bloomFilterWriteSupport.addKey(recordKey));
   }
 
+  @FunctionalInterface
+  private interface ValueWriter {
+    void write(SpecializedGetters row, int ordinal);
+  }
+
+  private void consumeMessage(Runnable writer) {
+    recordConsumer.startMessage();
+    writer.run();
+    recordConsumer.endMessage();
+  }
+
+  private void consumeGroup(Runnable writer) {
+    recordConsumer.startGroup();
+    writer.run();
+    recordConsumer.endGroup();
+  }
+
+  private void consumeField(String field, int index, Runnable writer) {
+    recordConsumer.startField(field, index);
+    writer.run();
+    recordConsumer.endField(field, index);
+  }
+
+  private void writeFields(InternalRow row, StructType schema, ValueWriter[] fieldWriters) {
+    for (int i = 0; i < row.numFields(); i++) {
+      int index = i;
+      if (!row.isNullAt(i)) {
+        StructField field = schema.fields()[i];
+        consumeField(field.name(), index, () -> fieldWriters[index].write(row, index));
+      }
+    }
+  }
+
+  private ValueWriter makeWriter(Schema avroSchema, DataType dataType) {
+    Schema resolvedSchema = avroSchema == null ? null : resolveNullableSchema(avroSchema);
+    LogicalType logicalType = resolvedSchema != null ? resolvedSchema.getLogicalType() : null;
+
+    if (dataType == DataTypes.BooleanType) {
+      return (row, ordinal) -> recordConsumer.addBoolean(row.getBoolean(ordinal));
+    } else if (dataType == DataTypes.DateType) {
+      return (row, ordinal) -> recordConsumer.addInteger((Integer) dateRebaseFunction.apply(row.getInt(ordinal)));
+    } else if (dataType == DataTypes.ShortType) {
+      return (row, ordinal) -> recordConsumer.addInteger(row.getShort(ordinal));
+    } else if (dataType == DataTypes.IntegerType) {
```

Comment:
I see `ParquetWriteSupport` also handles `YearMonthIntervalType` for integers.  Not sure if this class need to do that.